src/main/java/uk/gov/pay/adminusers/resources/InviteResource.java

```diff
@@ -129,6 +131,27 @@ public Invite updateInvite(@Parameter(example = "d02jddeib0lqpsir28fbskg9v0rv")
                 .collect(Collectors.toList());
         return inviteService.updateInvite(inviteCode, updateRequests);
     }
+    
+    @POST
+    @Path("/v1/api/invites/{code}/send-otp")
+    @Produces(APPLICATION_JSON)
+    public Response sendOtp(@Parameter(example = "d02jddeib0lqpsir28fbskg9v0rv") @PathParam("code") String inviteCode) {
+        LOGGER.info("Invite send OTP POST request for code - [ {} ]", inviteCode);
+
+        if (isNotBlank(inviteCode) && inviteCode.length() > MAX_LENGTH_CODE) {
+            throw new WebApplicationException(Response.Status.NOT_FOUND);
+        }
+
+        return inviteService.findInvite(inviteCode).map(inviteEntity -> {
```

Comment:
Finding the invite and checking whether there's a telephone number is probably better happening in the InviteService. We like to keep the resource as slim as possible, and have it mainly detailing with parsing the request and constructing the response.

Reply:
Yeah, I wasn't keen doing this stuff in the resource, particularly checking the telephone number.  Your way makes a lot more sense.